### PR TITLE
cxx-qt-gen: remove gen_cpp and use new phases for C++ side

### PR DESCRIPTION
**Pull request by ahayzen-kdab**
Depends on #248

### CODE REVIEW

File: cxx-qt-gen/src/generator/cpp/signal.rs [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+        // Generate the Q_SIGNAL
+        let parameter_types = signal
+            .parameters
+            .iter()
+            .map(|parameter| {
+                Ok(format!(
+                    "{cpp_ty} {ident}",
+                    ident = parameter.ident,
+                    cpp_ty = if let Some(cpp_ty) = &parameter.cpp_type {
```
Comment (ahayzen-kdab) [inline cxx-qt-gen/src/generator/cpp/signal.rs:39]: can this be common to the types.rs ?

File: crates/cxx-qt-gen/src/generator/cpp/signal.rs [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+
+        generated.signals.push(format!(
+            "void {ident}({parameters});",
+            ident = signal_ident,
+            parameters = parameter_types
+        ));
+
+        // Generate the queued emitters
+        //
+        // TODO: perform this in one loop
```
Comment (ahayzen-kdab) [inline crates/cxx-qt-gen/src/generator/cpp/signal.rs:53]: should be able to move all of these into one loop over the parameters.

File: crates/cxx-qt-gen/src/generator/cpp/fragment.rs [hunk 3]
```diff
@@ -8,3 +8,8 @@ pub struct CppFragmentPair {
     pub header: String,
     pub source: String,
 }
+
+pub struct CppNamedType {
+    pub ident: String,
+    pub ty: String,
```
Comment (LeonMatthesKDAB) [inline crates/cxx-qt-gen/src/generator/cpp/fragment.rs:14]: Should this `ty` be a CxxType?

Comment (LeonMatthesKDAB) [reply crates/cxx-qt-gen/src/generator/cpp/fragment.rs:14]: Also, regarding naming, CppNamedType or CxxNamedType?
We should choose one for our internal APIs.
Atm, we're always using Cpp, so maybe change CxxType to CppType?

Comment (ahayzen-kdab) [reply crates/cxx-qt-gen/src/generator/cpp/fragment.rs:14]: Regarding teh Cpp vs Cxx naming i thought that too. But maybe it's better to do a mass rename at the end as we already have a mixture of cpp and cxx module names etc etc?

File: crates/cxx-qt-gen/src/generator/cpp/invokable.rs [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+
+                    // If the name of the argument is self then ignore,
+                    // as this is likely the self: Pin<T>
+                    if ident == "self" {
+                        Ok(None)
+                    } else {
+                        Ok(Some(CppNamedType {
+                            ident: ident.to_string(),
+                            ty: format!(
+                                "{ty} {ident}",
```
[the remark this answers is not in the document]
Comment (ahayzen-kdab) [reply crates/cxx-qt-gen/src/generator/cpp/invokable.rs]: Removed :-)

File: crates/cxx-qt-gen/src/generator/cpp/fragment.rs (hunk 3 above)
Comment (ahayzen-kdab) [reply crates/cxx-qt-gen/src/generator/cpp/fragment.rs]: Changed to be CxxType :-)

Comment (ahayzen-kdab) [reply crates/cxx-qt-gen/src/generator/cpp/fragment.rs]: And renamed things to CppType so we use Cpp for now.

File: crates/cxx-qt-gen/src/generator/cpp/property/mod.rs [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+use syn::Result;
+
+mod emitter;
+mod getter;
+mod meta;
+mod setter;
+mod signal;
+
+pub fn generate_cpp_properties(
+    generated: &mut GeneratedCppQObjectBlocks,
```
[the remark this answers is not in the document]
Comment (ahayzen-kdab) [reply crates/cxx-qt-gen/src/generator/cpp/property/mod.rs]: As discussed changed :-)

**Review by LeonMatthesKDAB — APPROVED** [review]
LGTM

**Pull request merged**